presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -802,6 +821,21 @@ private static int validateValueIsPowerOfTwo(Object value, String property)
         return intValue;
     }
 
+    private static Integer validatePositiveIntegerValue(Object value, String property)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        int intValue = ((Number) value).intValue();
+        if (intValue <= 0) {
+            throw new PrestoException(
+                    INVALID_SESSION_PROPERTY,
+                    format("%s must be a positive integer: %s", property, intValue));
```

Comment:
I would imagine many other integer properties would choke on negative numbers. How do they do it?

Reply:
I just check the code. For existing integer properties, we either don't check the sign at all (e.g. `HASH_PARTITION_COUNT`, `QUERY_MAX_STAGE_COUNT`, etc.), or we do a `checkArgument()` when getting the session property value (e.g. `CONCURRENT_LIFESPANS_PER_NODE`, `QUERY_PRIORITY`, etc). I think it worth a separate commit to add non-negative/positive checks to all of these integer properties if necessary. I will add it in this PR later.

Reply:
I realized that the change will be larger than I expected, will create a separate PR about it. 

My current plan is to enforce a lower bound check and a null check in `PropertyMetadata.java::integerProperty()`. Similar checks can be done in other data types as well.